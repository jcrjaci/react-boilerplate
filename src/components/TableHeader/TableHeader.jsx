import React from 'react';
import PropTypes from 'prop-types';
import './TableHeader.scss';

/** Table component
* @param {object} props Component props
* @param {array} props.headers List of headers to be displayed in table header.
*/
const TableHeader = ({ headers }) => (
  <div className="table-header">
    {headers.map(field => <div key={field}>{field}</div>)}
  </div>
);

TableHeader.propTypes = {
  headers: PropTypes.arrayOf(PropTypes.string),
};

TableHeader.defaultProps = {
  headers: [],
};

export default TableHeader;
