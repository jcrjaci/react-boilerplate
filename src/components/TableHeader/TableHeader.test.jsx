import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';
import TableHeader from './TableHeader';

configure({ adapter: new Adapter() });

describe('<TableHeader />', () => {
  const headers = ['header1', 'header2'];
  const wrapper = shallow(<TableHeader headers={headers} />);

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render empty table header', () => {
    wrapper.setProps({ headers: [] });
    const tableHeader = wrapper.find('.table-header').children();
    expect(tableHeader).toHaveLength(0);
  });

  it('should render 4 headers', () => {
    wrapper.setProps({ headers: [...headers, 'header3', 'header4'] });
    const tableHeader = wrapper.find('.table-header').children();
    expect(tableHeader).toHaveLength(4);
    expect(tableHeader.at(0).key()).toBe('header1');
    expect(tableHeader.at(1).key()).toBe('header2');
    expect(tableHeader.at(2).key()).toBe('header3');
    expect(tableHeader.at(3).key()).toBe('header4');
  });
});
