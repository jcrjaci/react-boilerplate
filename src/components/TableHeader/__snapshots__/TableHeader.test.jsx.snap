// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableHeader /> should render correctly 1`] = `
<div
  className="table-header"
>
  <div
    key="header1"
  >
    header1
  </div>
  <div
    key="header2"
  >
    header2
  </div>
</div>
`;
